searx/engines/odysee.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Odysee_ is a decentralised video hosting platform.
+
+.. _Odysee: https://github.com/OdyseeTeam/odysee-frontend
+"""
+
+import time
+from datetime import datetime
+
+# Engine metadata
+about = {
+    "website": "https://odysee.com/",
+    "wikidata_id": "Q102046570",
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+# Engine configuration
+paging = True
+results_per_page = 20
+categories = ['images']
+
+# Search URL (Note: lighthouse.lbry.com/search works too, and may be faster at times)
+base_url = "https://lighthouse.odysee.tv/search"
+
+# Request function
+def request(query, params):
+    page = params.get("pageno", 1) - 1
+    start_index = page * results_per_page
+
+    query_params = {
+        "s": query,
+        "size": results_per_page,
+        "from": start_index,
+        "include": "channel,thumbnail_url,title,description,duration,release_time",
+        "mediaType": "video",
+    }
```

Comment:
From where did you get this REST API? .. I looked at the [odysee-frontend](https://github.com/OdyseeTeam/odysee-frontend) .. but I just could not find an API description in a hurry .. if you have more information, let us know and we can add it to the documentation.

Reply:
the only info i was able to find on this was posted by searx users here: https://github.com/searx/searx/issues/2504

based on this, it should be possible to add support for language filtering as well as safe search.

Reply:
> based on this, it should be possible to add support for language filtering as well as safe search.

would you like to add a [feature-request](https://github.com/searxng/searxng/issues/new?assignees=&labels=enhancement&projects=&template=feature-request.md&title=) .. just to have it on our *ToDo* list and not to forget / thanks!